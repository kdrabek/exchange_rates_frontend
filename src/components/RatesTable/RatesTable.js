import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table } from 'react-bootstrap';

class RatesTable extends Component {

  prepareHeader(headers){
    const columnNames = headers.map((header, index) => <th key={index}>{header}</th>);
    return (
      <thead>
        <tr>
          {columnNames}
        </tr>
      </thead>);

  }

  prepareBody(data){
    const tableBody = data.map((rowData, index) => {
      const row = rowData.map(column => <td>{column}</td>)
      return <tr key={index}>{row}</tr>;
    });

    return (
      <tbody>
        {tableBody}
      </tbody>
    );
  }

  render() {
    const headerRow = this.prepareHeader(this.props.headers || []);
    const bodyRow = this.prepareBody(this.props.data || [])
    return (
      <Table striped hover>  
        {headerRow}
        {bodyRow}
      </Table>
    );
  }
}

RatesTable.propTypes = {
  headers: PropTypes.array,
  data: PropTypes.array
};

export default RatesTable;