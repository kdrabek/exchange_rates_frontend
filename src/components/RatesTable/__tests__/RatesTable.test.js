import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import RatesTable from '../RatesTable';

describe('RatesTable component', () => {

  it('matches the snapshot when empty data', () => {
    const component = renderer.create(
      <RatesTable 
        headers={['header1', 'header2', 'header3']}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot when data passed', () => {
    const data = [
      ['row1_column1', 'row1_column2', 'row1_column2'], 
      ['row2_column1', 'row2_column2', 'row2_column2']
    ];
    const component = renderer.create(
      <RatesTable 
        headers={['header1', 'header2', 'header3']}
        data={data}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});