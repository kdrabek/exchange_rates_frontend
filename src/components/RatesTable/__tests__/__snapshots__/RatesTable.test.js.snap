exports[`RatesTable component matches the snapshot when data passed 1`] = `
<table
  className="table table-striped table-hover">
  <thead>
    <tr>
      <th>
        header1
      </th>
      <th>
        header2
      </th>
      <th>
        header3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        row1_column1
      </td>
      <td>
        row1_column2
      </td>
      <td>
        row1_column2
      </td>
    </tr>
    <tr>
      <td>
        row2_column1
      </td>
      <td>
        row2_column2
      </td>
      <td>
        row2_column2
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`RatesTable component matches the snapshot when empty data 1`] = `
<table
  className="table table-striped table-hover">
  <thead>
    <tr>
      <th>
        header1
      </th>
      <th>
        header2
      </th>
      <th>
        header3
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
